src/main/java/dev/nmarulo/depensaapp/security/JwtAuthenticationEntryPoint.java

```diff
@@ -0,0 +1,50 @@
+package dev.nmarulo.depensaapp.security;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import dev.nmarulo.depensaapp.commons.dtos.ErrorRes;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ProblemDetail;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.oauth2.jwt.JwtException;
+import org.springframework.security.oauth2.server.resource.web.BearerTokenAuthenticationEntryPoint;
+import org.springframework.security.web.AuthenticationEntryPoint;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.net.URI;
+
+@Component

```

Comment:
Is it necessary for the class to be a component?
I think it is not necessary considering the way it is implemented, similar to the `CustomJwtAuthenticationConverter` class.

Reply:
It is not in this case and can be done just like `CustomJwtAuthenticationConverter ` class,  but it is a good practice to keep some of the concerns separate, for instance the `ObjectMapper` bean which is injected in the `JwtAuthenticationEntryPoint` class bean is unique to the `JwtAuthenticationEntryPoint` class itself and injecting this bean inside the `WebSecurityConfig` class then passing it on to the `JwtAuthenticationEntryPoint` class would be a redundancy and would break the separation of concern concept as the `WebSecurityConfig` class itself has no use of this bean. 

In an ideal situation we would have directly injected both the `CustomJwtAuthenticationConverter` class (given that we turn it into a bean) and the `JwtAuthenticationEntryPoint` class to the `filterChain` method and it would have looked something like this:
```java
    @Bean
    public SecurityFilterChain filterChain(HttpSecurity http,
                                           CustomJwtAuthenticationConverter converter,
                                           JwtAuthenticationEntryPoint jwtAuthenticationEntryPoint) throws Exception {
        
        http.securityMatcher(appProperties.getPathPrefix() + "/**")
            .authorizeHttpRequests(authorize -> {
                if (ArrayUtils.isNotEmpty(appProperties.getPermitAllPaths())) {
                    authorize.requestMatchers(appProperties.getPermitAllPaths())
                             .permitAll();
                }
                
                authorize.anyRequest()
                         .authenticated();
            })
            .httpBasic(AbstractHttpConfigurer::disable)
            .csrf(AbstractHttpConfigurer::disable)
            .sessionManagement(session -> session.sessionCreationPolicy(SessionCreationPolicy.STATELESS))
            .oauth2ResourceServer(oauth -> oauth.authenticationEntryPoint(jwtAuthenticationEntryPoint)
                                                .jwt(jwtConfigurer -> jwtConfigurer.jwtAuthenticationConverter(converter)))
            .cors(value -> value.configurationSource(corsConfigurationSource()));
        
        return http.build();
    }
```
This way we could have gotten rid of both the `configOAuth2` method and the injected `UserRepository` and `LocalMessage` beans from the `WebSecurityConfig` class.

Hence, preferring to keep the `JwtAuthenticationEntryPoint` class as a bean for now. Please lmk what you think.

Reply:
Yes, it makes sense.

Ok, you can leave it like that.

I will aim to make this change so that the “Converter” dependencies are passed from the method arguments and not injecting them as class fields.

Thank you very much for the comments.